Updates Rule for Haiku

saloniig opened the pull request:
- Removes extra space between new and (

saloniig committed:
Updates Rule for Haiku
	- Removes extra space between new and (

pulkomandy commented on clang/lib/Format/TokenAnnotator.cpp:
```diff
@@ -2957,7 +2957,27 @@ bool TokenAnnotator::spaceRequiredBetween(const AnnotatedLine &Line,
       return false;
-    return Line.Type == LT_ObjCDecl || Left.is(tok::semi) ||
+
+    if(Style.BreakConstructorInitializers == FormatStyle::BCIS_Haiku){
```
Hi!

I think BreakConstructorInitializers is not the right variable to use here. Its documentation (https://clang.llvm.org/docs/ClangFormatStyleOptions.html) shows it is used for a different thing. It would be a bit confusing to have it affect other unrelated things.

Rather, a new "SpaceBeforeParens" style should be added (maybe named FormatStyle::SBPO_Haiku) and used here.

By cleanly separating all our changes under different options, it will be easier for other users of clang-format or haiku-format to reuse only some parts of our style, so it results in more flexibility for everyone.

saloniig replied:
Hello,
Thank you for your review. I have made the changes according to the SpaceBeforeParens style.
Please review. 
Thanks

pulkomandy replied:
Much better now, thanks!

saloniig committed:
Adds SBPO_Haiku clang-format
	- Adds SBPO_Haiku option in spaceBeforeParensOptions
	- Adds FormatStyle::SBPO_Haiku in getHaikuStyle
	- Updates BreakConstructorInitializers condition of style and BCIS_Haiku to SpaceBeforeParens SBPO_Haiku style in TokenAnnotator.cpp

pulkomandy commented on clang/lib/Format/Format.cpp:
```diff
@@ -1279,6 +1279,8 @@ Style.AccessModifierOffset = -4;
   Style.UseTab = FormatStyle::UT_Always;
   Style.SpaceAfterCStyleCast = false;
   Style.SpaceAfterTemplateKeyword = false;
+  Style.SpaceBeforeParens = FormatStyle::SBPO_Haiku;
```
you should also allow to enable this spearately from the other Haiku settings, like it is done here for BreakConstructorInitializersStyle (at line 216):

https://github.com/viveris/llvm-project/commit/18915202515159c42ceea3761ee9c22f8f5ea156#diff-c28a58077d2d1b988c339635e2fa38c9344343c64185de804c35b38c6ee9276dR216

saloniig replied:
I have sent a commit by doing required changes. Please check.
Thanks.

pulkomandy commented:
If you want to continue working on this, I suggest using clang-format on various source files in Haiku code. It will make some changes to the files, that should then be reviewed against our coding guidelines (https://www.haiku-os.org/development/coding-guidelines/) to see if the tool is correct, or if it's making unwanted changes and it needs to be improved.

saloniig committed:
Updates Format.cpp

- Enable SBPO_Haiku from the other Haiku settings

saloniig commented:
Yes, I would definitely love to do this and I will give further report soon after checking the tool.
